bindings/experimental/distrdf/python/DistRDF/Backends/Base.py

```diff
@@ -193,9 +198,85 @@ def register_initialization(cls, fun, *args, **kwargs):
 
             **kwargs (dict): Keyword arguments used to execute the function.
         """
-        cls.initialization = partial(fun, *args, **kwargs)
-        fun(*args, **kwargs)
+        cls.initialization = partial(fun, *args, **kwargs)    
+        fun(*args, **kwargs) 
+     
+    # TODO and check - make sure this works correctly for the multiple separate declares   
+    @classmethod
+    def register_declaration(cls, declaration): 
+        
+        declaration_string = cls.declaration_str + declaration
+        
+        cls.declaration_str += declaration_string
+        code_to_declare_with_guards = "#ifndef CODE\n#define CODE\n{}\n#endif\n".format(declaration_string)
```

Comment:
Here we can just leave the `cls.declaration_str += declaration_string` and use the class attribute in the next line without the extra variable